atlasdb-client/src/test/java/com/palantir/atlasdb/table/description/AbstractSchemaTest.java

```diff
@@ -0,0 +1,193 @@
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.SortedMap;
+import java.util.TreeMap;
+
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mockito;
+
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Lists;
+import com.google.common.primitives.UnsignedBytes;
+import com.palantir.atlasdb.encoding.PtBytes;
+import com.palantir.atlasdb.keyvalue.api.Cell;
+import com.palantir.atlasdb.keyvalue.api.ColumnSelection;
+import com.palantir.atlasdb.keyvalue.api.Namespace;
+import com.palantir.atlasdb.keyvalue.api.RangeRequest;
+import com.palantir.atlasdb.keyvalue.api.RowResult;
+import com.palantir.atlasdb.keyvalue.api.TableReference;
+import com.palantir.atlasdb.ptobject.EncodingUtils;
+import com.palantir.atlasdb.transaction.api.Transaction;
+import com.palantir.atlasdb.transaction.impl.AbstractTransaction;
+import com.palantir.common.base.BatchingVisitableFromIterable;
+
+
+public abstract class AbstractSchemaTest {
+    private static final TableReference tableRef =
+            TableReference.create(Namespace.DEFAULT_NAMESPACE, "GenericTest");
+
+    private static final String testRowKey = "testRowKey";
+    private static final String anotherTestRowKey = "testRowKey2";
+    private static final long testValue = 2L;
+    private static final long anotherTestValue = 3L;
+    private static ColumnSelection firstColSelection =
+            ColumnSelection.create(Collections.singletonList(PtBytes.toCachedBytes("c")));
+
+    protected static final String firstColShortName = "c";
+    protected static final String secondColShortName = "d";
+
+    protected abstract void putSingleRowFirstColumn(Transaction transaction, String roWKey, long value);
+    protected abstract Long getSingleRowFirstColumn(Transaction transaction, String rowKey);
+    protected abstract List<Long> getMultipleRowsFirstColumn(Transaction transaction, List<String> rowKey);
+    protected abstract List<String> getRangeSecondColumn(Transaction transaction, String startRowKey, String endRowKey);
+    protected abstract void deleteWholeRow(Transaction transaction, String rowKey);
+    protected abstract void deleteFirstColumn(Transaction transaction, String rowKey);
+
+    @Test
+    public void testPutSingleRowFirstColumn() {
+        AbstractTransaction transaction = mock(AbstractTransaction.class);
+
+        putSingleRowFirstColumn(transaction, testRowKey, testValue);
+
+        ArgumentCaptor<Map> argument = ArgumentCaptor.forClass(Map.class);
+        Mockito.verify(transaction, times(1)).put(eq(tableRef), argument.capture());
+
+        Cell expectedCell = Cell.create(PtBytes.toBytes(testRowKey), PtBytes.toBytes(firstColShortName));
+        assertEquals(argument.getValue().entrySet().size(), 1);
+        assertEquals(argument.getValue().keySet().toArray()[0], expectedCell);
+        assertTrue(Arrays.equals((byte[]) argument.getValue().values().toArray()[0],
```

Comment:
nit: We can use equals with a custom comparator `Arrays.equals` here in the interest of better messages if asserts fail. Same for all assertTrue.. we can explore using assertj here as that has a better syntax for such comparisions.